### PR TITLE
Update comments to resolve doxygen warnings

### PR DESCRIPTION
**Pull request by shariarriday**
Closes https://github.com/actor-framework/actor-framework/issues/1946

### CODE REVIEW

File: libcaf_core/caf/detail/parse.hpp [hunk 1]
```diff
@@ -43,30 +43,40 @@ struct literal {
   std::string_view str;
 };
 
+/// @memberof caf::detail
```
Comment (Neverlord) [inline libcaf_core/caf/detail/parse.hpp:46]: Why is this necessary? The function is properly defined in the namespace.

File: libcaf_core/caf/expected.hpp [hunk 2]
```diff
@@ -254,57 +254,53 @@ class expected {
...
 
-  /// @copydoc cvalue
+  /// @copydoc value
   const T* operator->() const noexcept {
     return &value_;
   }
 
   // -- error access -----------------------------------------------------------
 
-  /// @copydoc cerror
```
Comment (Neverlord) [inline libcaf_core/caf/expected.hpp:289]: This leaves them undocumented now.

File: libcaf_core/caf/flow/observable_builder.hpp [hunk 3]
```diff
@@ -78,15 +78,15 @@ class CAF_CORE_EXPORT observable_builder {
 
   observable_builder& operator=(const observable_builder&) noexcept = default;
 
-  /// Creates a @ref generation that emits values by repeatedly calling
-  /// `generator.pull(...)`.
+  /// Creates a \link caf::flow::generation \endlink that emits values by
+  /// repeatedly calling `generator.pull(...)`.
```
Comment (Neverlord) [inline libcaf_core/caf/flow/observable_builder.hpp:82]: Please stick to javadoc style, i.e., `@` prefix. Same for the other occurrences in this file.

File: libcaf_core/caf/flow/subscription.hpp [hunk 4]
```diff
@@ -108,6 +108,7 @@ class CAF_CORE_EXPORT subscription {
 
     bool disposed() const noexcept override;
 
+    /// Demand for `n` more items.
```
Comment (Neverlord) [inline libcaf_core/caf/flow/subscription.hpp:111]:
```suggestion
    /// Signals demand for `n` more items.
```

File: libcaf_core/caf/make_copy_on_write.hpp [hunk 5]
```diff
@@ -9,11 +9,12 @@
...
-/// @relates ref_counted
-/// @relatealso intrusive_cow_ptr
+/// @relates intrusive_cow_ptr
 template <class T, class... Ts>
 intrusive_cow_ptr<T> make_copy_on_write(Ts&&... xs) {
   return intrusive_cow_ptr<T>(new T(std::forward<Ts>(xs)...), false);
 }
 
+/// @relatesalso ref_counted
+
```
Comment (Neverlord) [inline libcaf_core/caf/make_copy_on_write.hpp:19]: Leftover? Nothing follows this line.

File: libcaf_core/caf/mixin/requester.hpp [hunk 6]
```diff
@@ -23,7 +23,9 @@
 namespace caf::mixin {
 
 /// A `requester` is an actor that supports
+/// @code
 /// `self->request(...).{then|await|receive}`.
+/// @endcode
```
Comment (Neverlord) [inline libcaf_core/caf/mixin/requester.hpp:28]: Why add the `code` block here?

File: libcaf_core/caf/telemetry/metric_registry.cpp
Comment (Neverlord) [inline libcaf_core/caf/telemetry/metric_registry.cpp:1]: Changes in this file are not only relevant to documentation. Please file as separate PR.

File: libcaf_core/caf/typed_actor_view_base.hpp [hunk 7]
```diff
@@ -6,7 +6,7 @@
 
 namespace caf {
 
-/// Tag type for @ref typed_actor_view.
+/// Tag type for \link caf::typed_actor_view \endlink.
```
Comment (Neverlord) [inline libcaf_core/caf/typed_actor_view_base.hpp:9]: Wrong style.

File: libcaf_io/caf/io/basp/all.hpp [hunk 8]
```diff
@@ -125,4 +126,4 @@
 /// individual BASP message types can be found in the documentation
 /// of {@link message_type} below.
 ///
-/// ![](basp_sequence.png)
+/// ![](basp_sequence.png)
```
Comment (Neverlord) [inline libcaf_io/caf/io/basp/all.hpp:129]: Broken file ending.

File: libcaf_io/caf/io/network/interfaces.hpp [hunk 9]
```diff
@@ -20,6 +20,7 @@
 namespace caf::io::network {
 
 // {protocol => address}
+/// @memberof caf::io::network
```
Comment (Neverlord) [inline libcaf_io/caf/io/network/interfaces.hpp:23]: Like the other uses: it's already in the proper namespace?

**Review by Neverlord — CHANGES_REQUESTED** [review]
This PR conflates several disjoint things, which makes it hard to discuss them individually:

1. Actual syntax fixes, e.g., `test_coordinator::test_coordinator::clock => test_coordinator::clock`.
2. Fix for the non-existing `thread-safe` via new `threadsafe` alias.
3. Add missing documentation.
4. Surround implementations with `@cond PRIVATE` in C++ files. On that note: wouldn't it be a simpler fix to simply tell Doxygen to only look at the headers? We never add documentation to `.cpp` files.

Please factor out 1 and 2 and file them as separate PRs. Those changes are noncontroversial. Regarding 4, I think we can fully revert those changes and have a more robust solution in place (i.e., tell Doxygen to not look at the `.cpp` files).

Everything related to 3 can then stay in this PR after pushing out the unrelated bits.

File: libcaf_core/caf/mixin/requester.hpp (hunk 6 above)
Comment (shariarriday) [reply libcaf_core/caf/mixin/requester.hpp:28]: Doxygen expects a `code` block here for the code as it is in a separate line. Removing it shows a warning.

File: libcaf_core/caf/detail/parse.hpp (hunk 1 above)
Comment (shariarriday) [reply libcaf_core/caf/detail/parse.hpp]: It seems like doxygen can not parse the namespace and function properly without this command. I am looking into the cause in more details and prepare a separate PR for it.

File: libcaf_io/caf/io/network/interfaces.hpp [hunk 10]
```diff
@@ -20,6 +20,7 @@
 namespace caf::io::network {
 
 // {protocol => address}
+/// @memberof std::map
```
Comment (shariarriday) [inline libcaf_io/caf/io/network/interfaces.hpp:23]: This has been added to indicate that `std::map` is not a member of `caf::io::network`. It confuses the typedef and throws an warning otherwise.

**Review by Neverlord — APPROVED** [review]
Comments just FYI, I've already made the adjustments.
